### PR TITLE
Issue 6248 - fix fanalyzer warnings

### LINKED ISSUE #6248: Fix gcc fanalyzer issues
**Issue Description**
gcc -fanalyzer displays errors

**Package Version and Platform:**
 - Platform: Fedora
 - Package and version: main branch

1. **Steps to Reproduce**

podman run -ti --rm quay.io/389ds/devcontainer /bin/bash

git clone https://github.com/389ds/389-ds-base
cd 389-ds-base
sed -i 's/debug_cflags="-g3 -ggdb -gdwarf-5  -O0"/debug_cflags="-g3 -ggdb -gdwarf-5  -O0 -fanalyzer"/' configure.ac
./autogen.sh
./configure --enable-debug
make 2> stderr.log
grep Wanalyzer stderr.log

**Expected results**
Final grep should be empty

**Additional context**
PR #6246 fixed some of the issues but there are still some others

### PR DESCRIPTION
**Pull request by progier389**
This change remove all gcc -fanalyzer warnings.
Number of them are not that interesting (False/positive due to some limts of the analyzer and some possible crashes when running out of memory.
But some of these warnings where real concerns.

Issue #6248 

Reviewed by: @droideck (Thanks!)

### CODE REVIEW

File: ldap/servers/slapd/libglobs.c [hunk 1]
```diff
@@ -9195,15 +9195,15 @@ config_set_value(
     case CONFIG_SPECIAL_REFERRALLIST:
         /* referral list is already an array of berval* */
         if (value)
-            slapi_entry_attr_replace(e, cgas->attr_name, (struct berval **)*value);
+            slapi_entry_attr_replace(e, cgas->attr_name, *(struct berval ***)value);
```
Comment (droideck) [inline ldap/servers/slapd/libglobs.c:9201]: We still have a warning here:

`stack-based buffer over-read [CWE-126] [-Wanalyzer-out-of-bounds]`

Comment (droideck) [reply ldap/servers/slapd/libglobs.c:9201]: Will something like this calm it?

``` c
    /* referral list is already an array of berval* */
    if (value) {
        struct berval **referral_list = *(struct berval ***)value;
        if (referral_list) {
            slapi_entry_attr_replace(e, cgas->attr_name, referral_list);
        } else {
            slapi_entry_attr_set_charptr(e, cgas->attr_name, "");
        }
    } else {
        slapi_entry_attr_set_charptr(e, cgas->attr_name, "");
    }
    break;
 ```

**Review by droideck — APPROVED** [review]
LGTM! And the tests passed fully, which is nice to see!

**Comment by progier389** [comment]
A bit funny:  previous @droideck remark make me realize that I got different result
 when using  "-O2 -g -fanalyzer"  (some errors)
 than when using  "-DDEBUG -DMCC_DEBUG -g3 -ggdb -gdwarf-5  -O0  -fanalyzer"  (no errors)

File: ldap/servers/slapd/libglobs.c (hunk 1 above)
Comment (progier389) [reply ldap/servers/slapd/libglobs.c:9201]: FYI: I got 10 warnings for libglobs.c, I suspect it is again the false positive about following different branches in different  switches

**Pull request merged**